javaslang/src/main/java/javaslang/collection/BitMappedTrie.java

```diff
@@ -55,10 +55,29 @@ private static int treeSize(int branchCount, int depthShift) {
         return branchCount * fullBranchSize;
     }
 
-    BitMappedTrie<T> prepend(T leading) {
-        final int newSize = length() + 1;
+    static <T> BitMappedTrie<T> ofAll(Iterable<? extends T> iterable, int size) {
+        if (size == 0) { return empty(); }
+
+        Object[] array = (iterable instanceof Collection<?>) ? ((Collection<?>) iterable).toArray()
```

Comment:
That would be nice :)
Are you referring to the fact that `toArray` already copies everything? That's actually an optimization, since it enabled `System.arrayCopy` usage, which uses `SIMD` instructions to speed up copies.
Could you please try your idea out, I would be very happy if we could speed up things further :)